packages/webpack/src/utils/module-federation/typescript.ts

```diff
@@ -4,22 +4,26 @@ import { dirname, join } from 'path';
 import { workspaceRoot } from '@nx/devkit';
 
 let tsConfig: ParsedCommandLine;
-let tsPathMappings: ParsedCommandLine['options']['paths'];
+let tsPathMappings: Map<string, ParsedCommandLine['options']['paths']> =
+  new Map();
 
 export function readTsPathMappings(
   tsConfigPath: string = process.env.NX_TSCONFIG_PATH ?? getRootTsConfigPath()
 ): ParsedCommandLine['options']['paths'] {
-  if (tsPathMappings) {
-    return tsPathMappings;
+  if (tsPathMappings.has(tsConfigPath)) {
+    return tsPathMappings.get(tsConfigPath);
   }
 
   tsConfig ??= readTsConfiguration(tsConfigPath);
```

Comment:
You need to not cache this too.

Reply:
This is caching by path now? Unless you see some issue with it?

it’ll take the path to the tsconfig from the build process and use that to build and entry in the cache

Reply:
Oh wait I see it. Yes good catch